sql/hive/src/test/scala/org/apache/spark/sql/hive/InsertIntoHiveTableSuite.scala

```diff
@@ -494,4 +494,34 @@ class InsertIntoHiveTableSuite extends QueryTest with TestHiveSingleton with Bef
         spark.table("t").write.insertInto(tableName)
       }
   }
+
+  test(
+    """SPARK-20594: This is a walk-around fix to resolve a Hive bug. Hive requires that the
+      |staging directory needs to avoid being deleted when users set hive.exec.stagingdir
+      |under the table directory.""".stripMargin) {
```

Comment:
uh, because that is not to create a Hive table. How about simplifying the test case to?
```Scala
  test("SPARK-20594: hive.exec.stagingdir was deleted by Hive") {
    // Set hive.exec.stagingdir under the table directory without start with ".".
    withSQLConf("hive.exec.stagingdir" -> "./test") {
      withTable("test_table") {
        sql("CREATE TABLE test_table (key int)")
        sql("INSERT OVERWRITE TABLE test_table SELECT 1")
        checkAnswer(sql("SELECT * FROM test_table"), Row(1))
      }
    }
  }
```